api/PayrollProcessor.Core.Domain/Intrastructure/Operations/Commands/StranglerCommandDispatcher.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using Ardalis.GuardClauses;
+using CSharpFunctionalExtensions;
+using PayrollProcessor.Core.Domain.Intrastructure.Operations.Factories;
+
+namespace PayrollProcessor.Core.Domain.Intrastructure.Operations.Commands;
+
+/// <summary>
+/// TODO: Temporarily named after the Strangler Fig Pattern as this serves as an implementation of <see cref="ICommandDispatcher"/> migrating to CSharpFunctionalExtensions from LanguageExt.
```

Comment:
Have you done this kind of naming in a project before? I haven't but I like the idea of calling this type out explicitly. That way, you know it should eventually be renamed when it takes over.

Reply:
I haven't done this before but the approach was mentioned in Code That Fits In Your Head and I couldn't think of a better name so I figured calling it out explicitly would be best while the code is in migration state.